### PR TITLE
Adding std function to paddle frontend.

### LINKED ISSUE #15474: std

### PR DESCRIPTION
**Pull request by darleybarreto**
Close #15474

### CODE REVIEW

**Review by hmahmood24 — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/paddle/tensor/stat.py [hunk 1]
```diff
@@ -1 +1,11 @@
 # global
+import ivy
+from ivy.func_wrapper import with_unsupported_dtypes
+from ivy.functional.frontends.paddle.func_wrapper import (
+    to_ivy_arrays_and_back,
+)
+
+@with_unsupported_dtypes({"2.4.2 and below": ("float16", "bfloat16")}, "paddle")
+@to_ivy_arrays_and_back
+def std(input, axis=None, unbiased=True, keepdim=False, name=None):
```
[the remark this answers is not in the document]
Comment (darleybarreto) [reply ivy/functional/frontends/paddle/tensor/stat.py:10]: sure thing!

File: ivy/functional/frontends/paddle/tensor/stat.py [hunk 2]
```diff
@@ -5,8 +5,12 @@
     to_ivy_arrays_and_back,
 )
 
+@with_unsupported_dtypes({"2.4.2 and below": ("float16", "bfloat16")}, "paddle")
+@to_ivy_arrays_and_back
+def mean(x, axis=None, keepdims=False, out=None):
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/stat.py:10]: Can you also change the argument `input` to `x` in the test for this function?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_paddle_transforms.py [hunk 3]
```diff
@@ -1,3 +1,37 @@
...
+    test_flags,
+):
+    input_dtype, x, axis, correction = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=x[0],
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_paddle_transforms.py:33]: You'll need to change the argument name here as well

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @darleybarreto. I can see that there are some lint/formatting issues with your PR and also some other issues which are not allowing the tests to run. I've reviewed the PR and left a few comments for you to address. Kindly ask for a re-review once you're done. 👍🏼

File: ivy/functional/frontends/paddle/tensor/stat.py (hunk 2 above)
Comment (darleybarreto) [reply ivy/functional/frontends/paddle/tensor/stat.py:10]: Sure!

**Comment by darleybarreto** [comment]
> Hey @darleybarreto. I can see that there are some lint/formatting issues with your PR and also some other issues which are not allowing the tests to run. I've reviewed the PR and left a few comments for you to address. Kindly ask for a re-review once you're done. 👍🏼

Ah, sorry about forgetting to request your review! Do you know how can I run locally the lint to check what it's missing?

**Comment by hmahmood24** [comment]
> > Hey @darleybarreto. I can see that there are some lint/formatting issues with your PR and also some other issues which are not allowing the tests to run. I've reviewed the PR and left a few comments for you to address. Kindly ask for a re-review once you're done. 👍🏼
> 
> Ah, sorry about forgetting to request your review! Do you know how can I run locally the lint to check what it's missing?

No worries. You can use `pre-commit` to detect and reformat your PR everytime you commit. You can look at this [short guide](https://unify.ai/docs/ivy/overview/contributing/setting_up.html#pre-commit) to use it on your code. Let me know if you have any other questions

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Probably best to ask for a re-review once you're sure that the test passes all tests/[checks](https://github.com/unifyai/ivy/actions/runs/5155243501/jobs/9284676350?pr=15477#step:5:103) 😅

**Comment by darleybarreto** [comment]
> Probably best to ask for a re-review once you're sure that the function passes all tests/[checks](https://github.com/unifyai/ivy/actions/runs/5155243501/jobs/9284676350?pr=15477#step:5:103) sweat_smile

Sorry, I had a comment and decided to delete it after asking for review :(

**Comment by darleybarreto** [comment]
There are a few commits ~~(trying to fix)~~ fixing lints, probably best to squash them before merging.

**Comment by darleybarreto** [comment]
@hmahmood24 For some reason the `run_tests (1)` is failing.

**Comment by hmahmood24** [comment]
> @hmahmood24 For some reason the `run_tests (1)` is failing.

@darleybarreto The CI is not picking up your test and is trying to run the `paddle.mean` test instead. Can you kindly attach the output of your test here when you run it locally?

**Comment by darleybarreto** [comment]
I tried to run [this](https://github.com/unifyai/ivy/actions/runs/5155288505/workflow?pr=15477#L60) part and the line `python determine_tests.py master1` gives: 

```
ValueError: SHA b'e62b83b8d7b2783627ad84e67137ec86f54f1764' could not be resolved, git returned: b'e62b83b8d7b2783627ad84e67137ec86f54f1764 missing'
```

IS there a easier way to run all the tests?

**Comment by darleybarreto** [comment]
Also, perhaps I should place the test in `tst_paddle_stat.py`?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_paddle_transforms.py [hunk 4]
```diff
@@ -1,3 +1,38 @@
...
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        x=x[0],
+        dim=axis,
+        unbiased=bool(correction),
+        keepdim=keepdims,
+    )
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_vision/test_paddle_transforms.py:38]: The correct location for the test would be in `test_paddle_stat.py`

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Yeah I guess that's right. The test will have to be placed in the correct file

**Comment by hmahmood24** [comment]
> I tried to run [this](https://github.com/unifyai/ivy/actions/runs/5155288505/workflow?pr=15477#L60) part and the line `python determine_tests.py master1` gives:
> 
> ```
> ValueError: SHA b'e62b83b8d7b2783627ad84e67137ec86f54f1764' could not be resolved, git returned: b'e62b83b8d7b2783627ad84e67137ec86f54f1764 missing'
> ```
> 
> Is there a easier way to run all the tests?

You should use `pytest` to run your tests locally on your CLI. That's the easiest and most appropriate way to debug and run all your tests

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by hmahmood24** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**